snapshots/storage/metastore_test.go

```diff
@@ -198,75 +200,73 @@ var baseInfo = map[string]snapshots.Info{
-		t.Fatal("Expected not active error")
-	}
-	if !errdefs.IsFailedPrecondition(err) {
-		t.Fatalf("Expected not active error, got %+v", err)
-	}
+	t.Helper()
+	assert.Assert(t, err != nil)
+	assert.Assert(t, errdefs.IsFailedPrecondition(err), "got %+v", err)
 }
 
 func assertNotCommitted(t *testing.T, err error) {
-	if err == nil {
-		t.Fatal("Expected active error")
-	}
-	if !errdefs.IsInvalidArgument(err) {
-		t.Fatalf("Expected active error, got %+v", err)
-	}
+	t.Helper()
+	assert.Assert(t, err != nil)
+	assert.Assert(t, errdefs.IsInvalidArgument(err), "got %+v", err)
 }
 
 func assertExist(t *testing.T, err error) {
-	if err == nil {
-		t.Fatal("Expected exist error")
-	}
-	if !errdefs.IsAlreadyExists(err) {
-		t.Fatalf("Expected exist error, got %+v", err)
-	}
+	t.Helper()
+	assert.Assert(t, err != nil)
+	assert.Assert(t, errdefs.IsAlreadyExists(err), "got %+v", err)
 }
 
-func testGetInfo(ctx context.Context, t *testing.T, ms *MetaStore) {
+func testGetInfo(ctx context.Context, t *testing.T, _ *MetaStore) {
 	for key, expected := range baseInfo {
 		_, info, _, err := GetInfo(ctx, key)
-		if err != nil {
-			t.Fatalf("GetInfo on %v failed: %+v", key, err)
-		}
-		// TODO: Check timestamp range
-		info.Created = time.Time{}
-		info.Updated = time.Time{}
-		assert.Equal(t, expected, info)
+		assert.NilError(t, err, "on key %v", key)
+
+		assert.Check(t, is.Compare(expected, info, cmpSnapshotInfo), "on key %v", key)
 	}
 }
 
-func testGetInfoNotExist(ctx context.Context, t *testing.T, ms *MetaStore) {
+// compare snapshot.Info Updated and Created fields by checking they are
+// within a threshold of time.Now()
+var cmpSnapshotInfo = cmp.FilterPath(
+	func(path cmp.Path) bool {
+		field := path.Last().String()
+		return field == ".Created" || field == ".Updated"
+	},
+	cmp.Comparer(func(expected, actual time.Time) bool {
+		// cmp.Options must be symmetric, so swap the args
+		if actual.IsZero() {
+			actual, expected = expected, actual
+		}
+		// actual value should be within a few seconds of now
+		now := time.Now()
+		delta := now.Sub(actual)
+		threshold := 10 * time.Second
+		return delta > -threshold && delta < threshold
+	}))
+
+func testGetInfoNotExist(ctx context.Context, t *testing.T, _ *MetaStore) {
 	_, _, _, err := GetInfo(ctx, "active-not-exist")
 	assertNotExist(t, err)
 }
 
-func testWalk(ctx context.Context, t *testing.T, ms *MetaStore) {
+func testWalk(ctx context.Context, t *testing.T, _ *MetaStore) {
 	found := map[string]snapshots.Info{}
 	err := WalkInfo(ctx, func(ctx context.Context, info snapshots.Info) error {
 		if _, ok := found[info.Name]; ok {
 			return errors.Errorf("entry already encountered")
 		}
-		// TODO: Check time range
-		info.Created = time.Time{}
-		info.Updated = time.Time{}
 		found[info.Name] = info
 		return nil
 	})
-	if err != nil {
-		t.Fatalf("Walk failed: %+v", err)
-	}
-	assert.Equal(t, baseInfo, found)
+	assert.NilError(t, err)
+	assert.Assert(t, is.Compare(baseInfo, found, cmpSnapshotInfo))
```

Comment:
This removed `TODO` is an example of how `is.Compare()` can be used to compare complex types. The `cmpSnapshotInfo` provides a custom way to compare the timestamps in the structure.